authz_simple.go

```diff
@@ -26,31 +27,19 @@ const (
 // SimpleAuthz is a trivial authorizer, mostly ensuring
 // proper permission check for the 'create initial user' case.
 type SimpleAuthz struct {
-	jwth JWTHandler
-	l    *log.Logger
-}
-
-func NewSimpleAuthz(jwth JWTHandler, l *log.Logger) *SimpleAuthz {
-	return &SimpleAuthz{
-		jwth: jwth,
-		l:    l}
+	l *log.Logger
 }
 
 // Authorize makes SimpleAuthz implement the Authorizer interface.
-func (sa *SimpleAuthz) Authorize(token string, resource, action string) error {
+func (sa *SimpleAuthz) Authorize(token *jwt.Token, resource, action string) error {
 	// bypass checks for login
+	// for other resources - verify authz
 	if resource == ResourceLogin {
 		return nil
 	}
 
-	// for other resources - parse token
-	tok, err := sa.jwth.WithLog(sa.l).FromJWT(token)
-	if err != nil {
-		return authz.ErrAuthzTokenInvalid
-	}
-
 	// check correct scope for initial user creation
-	scope := tok.Claims.Scope
+	scope := token.Claims.Scope
```

Comment:
ok, we can:
- return the 'unauthorized' error
- panic explicitly, since this is in fact a programmer error (failed function precondition)
- let it panic on its own

what's our choice?